### ISSUE TITLE
compiling against python3 fails

### ISSUE THREAD

**Issue by junghans** [laristra/flecsale#39]
```
In file included from /home/christoph/work/flecsale/libsrc/utils/test/python_utils.cc:21:0:
/home/christoph/work/flecsale/libsrc/utils/python_utils.h: In function ‘void ale::utils::python_set_program_name(std::__cxx11::string)’:
/home/christoph/work/flecsale/libsrc/utils/python_utils.h:76:50: error: ‘Py_DecodeLocale’ was not declared in this scope
  auto program = Py_DecodeLocale(name.c_str(), NULL);
                                             ^
```

**Comment by charest** [comment]
Python 3 should work.  This looks to be a python version 3 to 3.5 issue.  I just need to fix the ifdefs instead of forcing python 2.

**Comment by junghans** [comment]
Now I get:
```
[ 97%] Building CXX object apps/maire_hydro/3d/CMakeFiles/maire_hydro_3d.dir/driver.cc.o
In file included from /home/christoph/work/flecsale/libsrc/utils/test/python_utils.cc:21:0:
/home/christoph/work/flecsale/libsrc/utils/python_utils.h: In function ‘void ale::utils::python_set_program_name(std::__cxx11::string)’:
/home/christoph/work/flecsale/libsrc/utils/python_utils.h:75:54: error: cannot convert ‘char*’ to ‘wchar_t*’ for argument ‘1’ to ‘void Py_SetProgramName(wchar_t*)’
   Py_SetProgramName( const_cast<char*>(name.c_str()) );
                                                      ^
make[2]: *** [CMakeFiles/utils.dir/build.make:159: CMakeFiles/utils.dir/libsrc/utils/test/python_utils.cc.o] Error 1
make[1]: *** [CMakeFiles/Makefile2:1636: CMakeFiles/utils.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....
```

**Comment by junghans** [comment]
Fixed in 26d620a2c6e19350f22b0a9891b28a80dafc6158.
